front/src/common/api/osrdEditoastApi.ts

```diff
@@ -1,136 +1,178 @@
-import { sortBy } from 'lodash';
+import { isNil, sortBy } from 'lodash';
 
 import {
   type GetLightRollingStockApiResponse,
   type GetSpritesSignalingSystemsApiResponse,
   generatedEditoastApi,
   type Property,
+  type TrainScheduleResult,
 } from './generatedEditoastApi';
 
 const formatPathPropertiesProps = (props: Property[]) =>
   props.map((prop) => `props[]=${prop}`).join('&');
 
-const osrdEditoastApi = generatedEditoastApi.enhanceEndpoints({
-  endpoints: {
-    getLightRollingStock: {
-      transformResponse: (response: GetLightRollingStockApiResponse) => ({
-        ...response,
-        results: sortBy(response?.results, ['metadata.reference', 'name']),
+const osrdEditoastApi = generatedEditoastApi
+  .injectEndpoints({
+    endpoints: (builder) => ({
+      getAllTimetableByIdTrainSchedules: builder.query<
+        TrainScheduleResult[],
+        { timetableId: number }
+      >({
+        queryFn: async ({ timetableId }, { dispatch }) => {
+          const pageSize = 25;
+          let page = 1;
+          let reachEnd = false;
+          const result: TrainScheduleResult[] = [];
+          while (!reachEnd) {
+            const promise = dispatch(
+              osrdEditoastApi.endpoints.getTimetableByIdTrainSchedules.initiate(
+                {
+                  id: timetableId,
+                  pageSize,
+                  page,
+                },
+                { forceRefetch: true, subscribe: false }
+              )
+            );
+            // need to unsubscribe on get call to avoid cache issue
+            const { data } = await promise;
+            if (data) result.push(...data.results);
+            reachEnd = isNil(data?.next);
+            page += 1;
+          }
+          return { data: result };
+        },
+        providesTags: (result) =>
+          result
+            ? [
+                ...result.map(({ id }) => ({ type: 'train_schedule' as const, id })),
+                { type: 'train_schedule', id: 'LIST' },
+              ]
+            : [{ type: 'train_schedule', id: 'LIST' }],
```

Comment:
We don't use fine-grained tags for train schedules: mutations don't invalidate the tags mentioned here. I think this should just be `providesTags: ['timetable']`